nvptx-run.c

```diff
@@ -263,29 +328,34 @@ This program has absolutely no warranty.\n",
     }
   }
 
-#if 0
-  /* Default seems to be 1 KiB stack, 8 MiB heap.  */
-  size_t stack, heap;
-  cuCtxGetLimit (&stack, CU_LIMIT_STACK_SIZE);
-  cuCtxGetLimit (&heap, CU_LIMIT_MALLOC_HEAP_SIZE);
-  printf ("stack %ld heap %ld\n", stack, heap);
-#endif
+  if (verbose)
+    {
+      size_t free_mem_update;
+      r = cuMemGetInfo (&free_mem_update, &dummy);
+      fatal_unless_success (r, "could not get free memory");
+      report_val (stderr, "Program args reservation (effective)",
+		  free_mem - free_mem_update);
```

Comment:
Doesn't this difference computation implicitly assume that nothing else is using the GPU concurrently?  (Which is a wrong assumption?)  Or, does every process/CUDA context always have available all the GPU memory -- I don't remember the details, and have not yet looked that up.